### ISSUE TITLE
Make `Jobs.run_async` compatible with iterations

### ISSUE THREAD

**Issue by onmyraedar** [expectedparrot/edsl#837]
Two key issues:
- `iterations` is not an attribute of Jobs, so we don't send it to the server alongside the other attributes like agents and models. 
- We're using the `Jobs.run_async` method to run remote inferences jobs, and it isn't taking an iterations parameter right now.

**Comment by johnjosephhorton** [comment]
So I wonder if easiest path would be to modify this coop function to take an `n` parameter (this in Jobs).  

```python
remote_job_creation_data = coop.remote_inference_create(
                self,
                description=remote_inference_description,
                status="queued",
            )
```
I think I'd prefer that to making `n` an attribute of the Jobs object, as it's seems reasonable to define a Jobs object before you know how many times you want to it done. What do you think @onmyraedar ? 

I presume on the coop side we're using the `JobsRunnerAsyncIO` to actually run?

**Comment by onmyraedar** [comment]
Yes, adding the parameter to `remote_inference_create` makes sense - we can do that!

And yes, we're calling `run_async`, which uses `JobsRunnerAsyncIO`.

**Comment by onmyraedar** [comment]
Closed with a20e80d
